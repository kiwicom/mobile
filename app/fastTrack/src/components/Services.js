// @flow strict

import * as React from 'react';
import { View } from 'react-native';
import { Text, StyleSheet, Icon, Translation } from '@kiwicom/mobile-shared';
import { type TranslationKeysType } from '@kiwicom/mobile-localization';
import { defaultTokens } from '@kiwicom/mobile-orbit';

type Service = {|
  +id: number,
  +icon: string,
  +translation: TranslationKeysType,
|};

type Props = {|
  +services: Service[],
|};

const Services = ({ services }: Props) => (
  <React.Fragment>
    <Text style={style.title}>
      <Translation id="mmb.fast_track.banner.services_label.bergamo.bergamo_stopover" />
    </Text>
    <View style={style.list}>
      {services.map((service: Service) => (
        <View style={style.item} key={service.id}>
          <Icon name={service.icon} color={defaultTokens.colorIconSecondary} />
          <Text style={style.name}>
            <Translation id={service.translation} />
          </Text>
        </View>
      ))}
    </View>
  </React.Fragment>
);

Services.defaultProps = {
  services: [
    {
      id: 1,
      icon: 'walk',
      translation: 'mmb.fast_track.banner.services_security_route.bergamo.bergamo_stopover',
    },
    {
      id: 2,
      icon: 'baggage-checked',
      translation: 'mmb.fast_track.banner.services_handling_assistance.bergamo.bergamo_stopover',
    },
  ],
};

export default Services;

const style = StyleSheet.create({
  title: {
    fontSize: 12,
    fontWeight: 'bold',
    textTransform: 'uppercase',
    color: defaultTokens.paletteInkLight,
  },
  list: {
    marginTop: 8,
    marginBottom: 16,
  },
  item: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 8,
  },
  name: {
    marginStart: 4,
    fontWeight: 'bold',
    color: defaultTokens.paletteInkNormal,
  },
});
