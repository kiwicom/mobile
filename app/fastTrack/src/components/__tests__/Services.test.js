// @flow

import React from 'react';
import renderer from 'react-test-renderer';

import Services from '../Services';

describe('Services', () => {
  test('Should render correctly', () => {
    const component = renderer.create(<Services />);
    expect(component).toMatchSnapshot();
  });
});
