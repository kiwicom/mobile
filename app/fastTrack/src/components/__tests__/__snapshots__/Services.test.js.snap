// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Services Should render correctly 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "#252A31",
          "fontFamily": "CircularPro-Book",
          "fontSize": 14,
          "fontWeight": "normal",
          "letterSpacing": -0.15,
        },
        Object {
          "color": "#5F738C",
          "fontSize": 12,
          "fontWeight": "bold",
          "textTransform": "uppercase",
        },
      ]
    }
  >
    <Text>
      Services included
    </Text>
  </Text>,
  <View
    style={
      Object {
        "marginBottom": 16,
        "marginTop": 8,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "orbit-icons",
            },
            Object {
              "fontSize": 24,
              "height": 24,
              "lineHeight": 24,
              "width": 24,
            },
            Object {
              "color": "#5F738C",
            },
            Array [
              undefined,
              null,
            ],
          ]
        }
      >
        ÓÅí
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#252A31",
              "fontFamily": "CircularPro-Book",
              "fontSize": 14,
              "fontWeight": "normal",
              "letterSpacing": -0.15,
            },
            Object {
              "color": "#252A31",
              "fontWeight": "bold",
              "marginStart": 4,
            },
          ]
        }
      >
        <Text>
          Fast track security route
        </Text>
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 8,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "orbit-icons",
            },
            Object {
              "fontSize": 24,
              "height": 24,
              "lineHeight": 24,
              "width": 24,
            },
            Object {
              "color": "#5F738C",
            },
            Array [
              undefined,
              null,
            ],
          ]
        }
      >
        h
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#252A31",
              "fontFamily": "CircularPro-Book",
              "fontSize": 14,
              "fontWeight": "normal",
              "letterSpacing": -0.15,
            },
            Object {
              "color": "#252A31",
              "fontWeight": "bold",
              "marginStart": 4,
            },
          ]
        }
      >
        <Text>
          Baggage handling assistance
        </Text>
      </Text>
    </View>
  </View>,
]
`;
